lerobot/common/robot_devices/utils.py

```diff
@@ -16,6 +16,20 @@ def busy_wait(seconds):
             time.sleep(seconds)
 
 
+def safe_disconnect(func):
+    # TODO(aliberts): Allow to pass custom exceptions
+    # (e.g. ThreadServiceExit, KeyboardInterrupt, SystemExit, UnpluggedError, DynamixelCommError)
+    def wrapper(robot, **kwargs):
+        try:
+            return func(robot, **kwargs)
+        except Exception as e:
+            if robot.is_connected:
+                robot.disconnect()
+            raise e
+
+    return wrapper
```

Comment:
Could you add an example use case for why safe_disconnect is needed?
The downside is that post mortem debugging wont work if you dont remove it (i.e. to breakpoint automatically at the location of the error).

Reply:
This wrapper is a good idea. The cameras, dxl busses, etc. drivers tend to sporadically throw exceptions (e.g. system has CPU load and a timeout is violated). Does `raise e` pass along the driver exception upward? The exceptions can be useful for debugging.